foreman/data_refinery_foreman/foreman/management/commands/correct_affy_cdfs.py

```diff
@@ -40,30 +44,36 @@ def handle(self, *args, **options):
 
         while True:
             for experiment in page.object_list:
-                gse = GEOparse.get_GEO(experiment.accession_code, destdir='/tmp/', how="brief", silent=True)
-
-                sample_accessions = list(gse.gsms.keys())
-                samples = Sample.objects.filter(accession_code__in=sample_accessions)
-
-                wrong_platform = False
-                for sample in samples:
-                    gpl = gse.gsms[sample.accession_code].metadata['platform_id'][0]
-                    internal_accession = get_internal_microarray_accession(gpl)
-                    if internal_accession != sample.platform_accession_code:
-                        wrong_platform = True
-                        break
-
-                if wrong_platform:
-                    if options["dry_run"]:
-                        logger.info("Would have re-surveyed experiment with accession code %s",
-                                    experiment.accession_code)
-                    else:
-                        logger.info("Re-surveying experiment with accession code %s",
-                                    experiment.accession_code)
-
-                        purge_experiment(experiment.accession_code)
-
-                        queue_surveyor_for_accession(experiment.accession_code)
+                try:
+                    gse = GEOparse.get_GEO(experiment.accession_code, destdir=GEO_TEMP_DIR, how="brief", silent=True)
+
+                    sample_accessions = list(gse.gsms.keys())
+                    samples = Sample.objects.filter(accession_code__in=sample_accessions)
+
+                    wrong_platform = False
+                    for sample in samples:
+                        gpl = gse.gsms[sample.accession_code].metadata['platform_id'][0]
+                        internal_accession = get_internal_microarray_accession(gpl)
+                        if internal_accession != sample.platform_accession_code:
+                            wrong_platform = True
+                            break
+
+                    if wrong_platform:
+                        if options["dry_run"]:
+                            logger.info("Would have re-surveyed experiment with accession code %s",
+                                        experiment.accession_code)
+                        else:
+                            logger.info("Re-surveying experiment with accession code %s",
+                                        experiment.accession_code)
+
+                            purge_experiment(experiment.accession_code)
+
+                            queue_surveyor_for_accession(experiment.accession_code)
+                finally:
+                    # GEOparse downloads files here and never cleans them up! Grrrr!
+                    download_path = GEO_TEMP_DIR + experiment.accession_code + '_family.soft.gz'
```

Comment:
Does GEO download this file for all experiments? I'm asking because in the GEO downloader we have to handle several types of files.

Reply:
I think that's because the downloader is dealing with more than just metadata.